### PR TITLE
Upgrade Spring Boot dependency to 2.5.13

### LINKED ISSUE #2010: Spring4shell Vulnerability.
### Description

Hi, guys.

Due to the spring4shell vulnerability [CVE-2022-22965](https://nvd.nist.gov/vuln/detail/CVE-2022-22965), we are updating the Spring Framework in our applications, after updating we noticed that some applications that use the Sentry Java SDK continued to point out the vulnerability, and upon investigating we realized that it is due to the Sentry SDK using a version of Spring Boot that is [vulnerable](https://github.com/getsentry/sentry-java/blob/main/buildSrc/src/main/java/Config.kt#L7) to spring4shell.

We tried to find an SDK version that used some version of Spring Boot with the fix, but we couldn't find it.

We are using version 5.1.0 of Sentry Java SDK and our applications use Java 11 + Spring.

More Information:

https://spring.io/blog/2022/03/31/spring-framework-rce-early-announcement

Are there any plans to upgrade the Spring Framework version used by the SDK?

Captura de Tela 2022-04-26 às 16 00 02
Captura de Tela 2022-04-26 às 16 03 30

If there was any mistake on my part, I apologize in advance.

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :scroll: Description

Upgrade Spring Boot dependency to 2.5.13

## :bulb: Motivation and Context

Fixes #2010

## :pencil: Checklist

- [x] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] No breaking changes

### CODE REVIEW

File: buildSrc/src/main/java/Config.kt [hunk 1]
```diff
@@ -4,9 +4,7 @@ object Config {
     val kotlinVersion = "1.5.31"
     val kotlinStdLib = "stdlib-jdk8"
 
-    val springBootVersion = "2.4.4"
-    // Spring is currently not compatible with Kotlin 1.4
-    val springKotlinCompatibleLanguageVersion = "1.3"
+    val springBootVersion = "2.5.13"
```
[the remark this answers is not in the document]
Comment (maciejwalkowiak) [reply buildSrc/src/main/java/Config.kt:7]: No, similar to Logback and Log4j - we do not dictate the Spring Boot version used in user projects (as explained here https://github.com/getsentry/sentry-java/issues/2010#issuecomment-1110192928). It only changes version used by us and as long as we don't use anything that got introduced in 2.5 we're good. Anyways, 2.4 reached end of life and all Spring Boot users are recommended to upgrade to >= 2.5

**Comment by adinauer** [comment]
Do we also want this for `5.7.x`?

**Review by adinauer — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ Starting with version `6.6.0` of the `sentry`, [Sentry's version >= v21.9.0](htt
 
 * Fix: Profiling rate decreased from 300hz to 100hz; fixed profiling traces folder creation on manual sdk init (#1997)
 * Fix: Allow disabling sending of client reports via Android Manifest and external options (#2007)
+* Ref: Upgrade Spring Boot dependency to 2.5.13 (#2011)
```
Comment (marandaneto) [inline CHANGELOG.md:9]: @maciejwalkowiak or @adinauer please also add it to https://docs.sentry.io/platforms/java/migration/

Comment (maciejwalkowiak) [reply CHANGELOG.md:9]: It does not require anything from users migrating to Sentry SDK 6.0

**Pull request merged**